packages/web3-provider/src/index.js

```diff
@@ -207,7 +207,11 @@ class WalletConnectProvider extends ProviderEngine {
   }
 
   async close () {
-    const wc = await this.getWalletConnector()
+    // Not using .getWalletConnector() there because if session is already
+    // killed (wc.connected is false), it would try to initialize a new session in the getter
+    // with an invalid wc.uri
+
+    const wc = this.wc
```

Comment:
We should check if wc exists and is connected. Perhaps this would be best if it were changed as follows:
```javascript
async close() {
    let wc
    if (this.wc && this.wc.connected) {
        wc = await this.getWalletConnector()
    } else if (this.wc) {
        wc = this.wc
    }
 
    if (wc) {
        await wc.killSession()
    }
    await this.stop()
    this.emit('close', 1000, 'Connection closed')
}
```

Reply:
Alternatively, we could pass an optional argument to `getWalletConnector` to disable the re-connection in case it's not connected

Reply:
@pedrouid I went with second solution because I thought that an ability to explicitly disable this behaviour for the getter would be better.